internal/resources/k6/resource_schedule.go

```diff
@@ -0,0 +1,450 @@
+package k6
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/grafana/k6-cloud-openapi-client-go/k6"
+
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.ResourceWithConfigure   = (*scheduleResource)(nil)
+	_ resource.ResourceWithImportState = (*scheduleResource)(nil)
+)
+
+var (
+	resourceScheduleName = "grafana_k6_schedule"
+	resourceScheduleID   = common.NewResourceID(common.IntIDField("id"))
+)
+
+func resourceSchedule() *common.Resource {
+	return common.NewResource(
+		common.CategoryK6,
+		resourceScheduleName,
+		resourceScheduleID,
+		&scheduleResource{},
+	)
+}
+
+// scheduleResourceModel maps the resource schema data.
+type scheduleResourceModel struct {
+	ID          types.String `tfsdk:"id"`
+	LoadTestID  types.String `tfsdk:"load_test_id"`
+	Starts      types.String `tfsdk:"starts"`
+	Frequency   types.String `tfsdk:"frequency"`
+	Interval    types.Int64  `tfsdk:"interval"`
+	Occurrences types.Int64  `tfsdk:"occurrences"`
+	Until       types.String `tfsdk:"until"`
+	Deactivated types.Bool   `tfsdk:"deactivated"`
+	NextRun     types.String `tfsdk:"next_run"`
+	CreatedBy   types.String `tfsdk:"created_by"`
+}
+
+// scheduleResource is the resource implementation.
+type scheduleResource struct {
+	basePluginFrameworkResource
+}
+
+// Metadata returns the resource type name.
+func (r *scheduleResource) Metadata(_ context.Context, _ resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = resourceScheduleName
+}
+
+// Schema defines the schema for the resource.
+func (r *scheduleResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages a k6 schedule for automated test execution.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Numeric identifier of the schedule.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"load_test_id": schema.StringAttribute{
+				Description: "The identifier of the load test to schedule.",
+				Required:    true,
+			},
```

Comment:
`load_test_id` should be immutable. Otherwise, changing it will create a schedule for another test but not remove the schedule for the old one.

```suggestion
			"load_test_id": schema.StringAttribute{
				Description: "The identifier of the load test to schedule.",
				Required:    true,
				PlanModifiers: []planmodifier.String{
					stringplanmodifier.RequiresReplace(),
				},
			},
```